cmake/kokkos_enable_options.cmake

```diff
@@ -48,6 +48,7 @@ KOKKOS_ENABLE_OPTION(CUDA_LAMBDA ${CUDA_LAMBDA_DEFAULT} "Whether to allow lambda
 # resolved but we keep the option around a bit longer to be safe.
 KOKKOS_ENABLE_OPTION(IMPL_CUDA_MALLOC_ASYNC ON  "Whether to enable CudaMallocAsync (requires CUDA Toolkit 11.2)")
 KOKKOS_ENABLE_OPTION(IMPL_NVHPC_AS_DEVICE_COMPILER OFF "Whether to allow nvc++ as Cuda device compiler")
+KOKKOS_ENABLE_OPTION(SYCL_NO_IMMEDIATE_COMMAND_LISTS OFF "Whether to disallow immediate command lists in the SYCL backend (necessary for Graph support)")
```

Comment:
Considering that this does not seem to be a long term solution, please make it an IMPL_ option